trappy/ftrace.py

```diff
@@ -191,46 +188,95 @@ def unregister_parser(cls, cobject):
             if cobject == obj:
                 del scope_classes[name]
 
-    def _do_parse(self):
-        params = {'window': self.window, 'abs_window': self.abs_window}
-        if not self.__class__.disable_cache and self._check_trace_cache(params):
-            # Read csv into frames
+    def _update_cache(self):
+        try:
+            # Recreate basic cache directories only if nothing cached
+            if not any([c.cached for c in self.trace_classes]):
+                self._create_trace_cache()
+
+            # Write out only events that weren't cached before
+            for trace_class in self.trace_classes:
+                if trace_class.cached:
+                    continue
+                csv_file = self._get_csv_path(trace_class)
+                trace_class.write_csv(csv_file)
+        except OSError as err:
+            warnings.warn(
+                "TRAPpy: Cache not created due to OS error: {0}".format(err))
+
+    def _max_window(self):
+        """
+        Compute the maximum window of the 'window' & 'abs_window' intersection
+        """
+        max_window = [0, None]
+        max_window[0] = max(self.window[0] + self.basetime, self.abs_window[0])
+
+        if (self.window[1] is not None) and (self.abs_window[1] is not None):
+            max_window[1] = max(self.window[1] + self.basetime, self.abs_window[1])
+        elif self.window[1] is not None:
+            max_window[1] = self.window[1] + self.basetime
+        elif self.abs_window[1] is not None:
+            max_window[1] = self.abs_window[1]
+
+        return max_window
+
+    def _windowify_class(self, trace_class, window):
+        if len(trace_class.data_frame) < 1:
+            return
+
+        if window[1]:
+            trace_class.data_frame = trace_class.data_frame[
+                window[0]:window[1]]
+        elif window[0]:
+            trace_class.data_frame = trace_class.data_frame[
+                window[0]:]
+
+    def _load_cache(self):
+        if self._check_trace_cache():
+            self.max_window = self._max_window()
+             # Read csv into frames
             for trace_class in self.trace_classes:
                 try:
                     csv_file = self._get_csv_path(trace_class)
                     trace_class.read_csv(csv_file)
+                    self._windowify_class(trace_class, self.max_window)
```

Comment:
Let's have a comment here explaining that the cache was written without the window applied, so we need to apply it after reading.